src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -60,6 +61,15 @@ class FileService(fileRepository: FileRepository,
             val pathWithoutSlash = if (m.originalPath.startsWith("/")) m.originalPath.tail else m.originalPath
             pathWithoutSlash == path
           }).head
+
+          //Add the 0 byte status here as know the file size at this point
+          //DROID does not identify 0 byte files therefore cannot set this status at the FFID stage
+          if (input.fileSize == 0) {
+            val statusRow = FilestatusRow(
+              uuidSource.uuid, fileId, FFID, ZeroByteFile, Timestamp.from(timeSource.now))
+            fileStatusRepository.addFileStatuses(List(statusRow))
```

Comment:
It's likely this can be a list of rows, as even for an individual file could be multiple statuses to add, for example FFID, Checksum etc. An individual file can also have multiple FFID statuses. 

So I would rather keep this as a list, what do you think?

Reply:
OK, let's leave as it is for now.